packages/use-contractkit/src/utils/metamask.ts

```diff
@@ -0,0 +1,233 @@
+}
+
+interface CeloTokens {
+  CELO: GoldTokenWrapper;
+  cUSD: StableTokenWrapper;
+  cEUR: StableTokenWrapper;
+}
+type StableTokens = Omit<CeloTokens, 'CELO'>;
+
+interface AddEthereumChainParameter {
+  chainId: string;
+  chainName: string;
+  nativeCurrency: Omit<ERC20Token, 'address'>;
+  rpcUrls: string[];
+  blockExplorerUrls?: string[];
+  iconUrls?: string[]; // Currently ignored by metamask
+}
+
+const NETWORKS = {
+  [ChainId.Mainnet]: Mainnet,
+  [ChainId.Alfajores]: Alfajores,
+  [ChainId.Baklava]: Baklava,
+};
+
+enum MetamaskRPCErrorCode {
+  AwaitingUserConfirmation = -32002,
+  UnknownNetwork = 4902,
+}
+
+interface MetamaskRPCError {
+  code: MetamaskRPCErrorCode;
+  message: string;
+  stack: string;
+}
+
+const makeNetworkParams = async (
+  info: Network,
+  CELO: GoldTokenWrapper
+): Promise<AddEthereumChainParameter> => {
+  const [symbol, decimals, name] = await Promise.all([
+    CELO.symbol(),
+    CELO.decimals(),
+    CELO.name(),
+  ]);
+
+  return {
+    chainId: `0x${info.chainId.toString(16)}`,
+    chainName: params[info.chainId].chainName ?? info.name,
+    nativeCurrency: {
+      name: params[info.chainId].nativeCurrency.name,
+      symbol,
+      decimals,
+    },
+    rpcUrls: [info.rpcUrl],
+    blockExplorerUrls: [info.explorer],
+    iconUrls: [`https://celoreserve.org/assets/tokens/${symbol}.svg`], // unfortunately unused
+  };
+};
+
+const tokenToParam = async (
+  token: GoldTokenWrapper | StableTokenWrapper
+): Promise<AddERC20TokenParameter> => {
+  const [symbol, decimals, name] = await Promise.all([
+    token.symbol(),
+    token.decimals(),
+    token.name(),
+  ]);
+
+  return {
+    type: 'ERC20',
+    options: {
+      address: token.address,
+      name,
+      symbol,
+      decimals,
+      image: `https://celoreserve.org/assets/tokens/${symbol}.svg`,
+    },
+  };
+};
+
+const makeAddCeloTokensParams = async (
+  tokens: StableTokens
+): Promise<AddERC20TokenParameter[]> =>
+  Promise.all(Object.values(tokens).map(tokenToParam));
+
+const addTokensToMetamask = async (
+  ethereum: Ethereum,
+  tokens: StableTokens
+) => {
+  const tokenParams = await makeAddCeloTokensParams(tokens);
+  const added = (
+    await Promise.all(
+      tokenParams.map((params) =>
+        ethereum?.request({
+          method: 'wallet_watchAsset',
+          params,
+        })
+      )
+    )
+  ).every(Boolean);
+
+  return added;
+};
+
+const addNetworkToMetamask = async (
+  ethereum: Ethereum,
+  networkConfig: Network
+): Promise<void> => {
+  const { CELO, ...tokens } = (await newKit(
+    networkConfig.rpcUrl
+  ).celoTokens.getWrappers()) as CeloTokens;
+
+  if (!CELO) {
+    throw new Error(
+      `Couldnt fetch CELO information for ${networkConfig.name}. Something's wrong`
+    );
+  }
+
+  try {
+    await ethereum?.request({
+      method: 'wallet_addEthereumChain',
+      params: [await makeNetworkParams(networkConfig, CELO)],
+    });
+
+    await addTokensToMetamask(ethereum, tokens);
```

Comment:
note: you'll notice when only add tokens automatically when adding the network to metamask, I didn't find a way to detect if a token is already added or not.